edb/server/pgcon/pgcon.pyx

```diff
@@ -435,9 +474,12 @@ cdef class PGConnection:
             self.msg_waiter = None
 
     def set_server(self, server):
-        assert defines.EDGEDB_SYSTEM_DB in self.dbname
         self.server = server
 
+    def mark_as_system_db(self):
+        assert defines.EDGEDB_SYSTEM_DB in self.dbname
```

Comment:
Why does this check use `in` and not `==`?

Reply:
It's related to multi-tenancy: https://github.com/edgedb/edgedb/issues/2158